app/main/views.py

```diff
@@ -104,21 +104,28 @@ def update(service_id, section):
         list(request.form.items()) + list(request.files.items())
     )
 
-    errors = Validate(content, service, posted_data, s3_uploader).errors
+    form = Validate(content, service, posted_data, s3_uploader)
+    update = {}
 
     for question_id in posted_data:
-        if question_id not in errors:
-            service_loader.set(service, question_id, "new value")
-
-    service_loader.post(service)
+        if question_id not in form.errors and question_id in form.clean_data:
```

Comment:
The side effects of the `errors` property are a bit non-obvious. For example, if these conditions are reversed it no longer works. Could the work of the errors property be done in an explicit validate function (perhaps called from the constructor)? If lazy properties is essential could the `clean_data` attribute be changed to a property that first calls `errors` to preload?